### PR TITLE
Implemented CountryNumber type

### PR DESCRIPTION
**Pull request by dejobratic**
- Created a CountryNumber value object with short as underlying type.
- CountryNumber can only be created by providing a number from the list of country numbers according to ISO-3166 standard.
- ToString method prints the three-digit format of the number.

### CODE REVIEW

**Review by andresharpe — APPROVED** [review]
Good coding! Thanks Mr. Hunt.

🚀

File: src/Nox.Types.EntityFramework/CountryNumberConverter.cs
Comment (rochar) [inline src/Nox.Types.EntityFramework/CountryNumberConverter.cs:1]: Can we move this class to a subFolder called ./Types/Month? keeping the namespace, and having the same folder struct in Nox.Types,NoxTypes.EntityFramework, and Nox.Types.Tests?

We are working on setting this conventions, maybe you can start following what will be propose, thks a lot

File: tests/Nox.Types.Tests/NoxCountryNumberTest.cs
Comment (rochar) [inline tests/Nox.Types.Tests/NoxCountryNumberTest.cs:1]: Can we move this class to a subFolder called ./Types/Month? keeping the namespace, and having the same folder struct in Nox.Types,NoxTypes.EntityFramework, and Nox.Types.Tests?

And rename the class to CountryNumberTest

We are working on setting this conventions, maybe you can start following what will be propose, thks a lot

File: tests/Nox.Types.Tests/NoxCountryNumberTest.cs [hunk 1]
```diff
@@ -0,0 +1,309 @@
+﻿namespace Nox.Types.Tests;
+
+public class NoxCountryNumberTest
+{
+    [Theory]
```
Comment (rochar) [inline tests/Nox.Types.Tests/NoxCountryNumberTest.cs:5]: the inline data is so huge, we could split it to a ClassData, and keep it in its own file, apart from the test class. see https://hamidmosalla.com/2017/02/25/xunit-theory-working-with-inlinedata-memberdata-classdata/

**Review by rochar — COMMENTED** [review]
Looks good, just left a few suggestion
Thks

**Pull request merged**